src/SIM/SIMoutput.C

```diff
@@ -517,45 +517,42 @@ int SIMoutput::writeGlvS1 (const Vector& psol, int iStep, int& nBlock,
   else if (!myVtf)
     return -99;
 
-  bool scalarEq = scalarOnly || this->getNoFields() == 1;
-  size_t nVcomp = scalarEq ? 1 : this->getNoFields();
-  if (nVcomp > this->getNoSpaceDim())
-    nVcomp = this->getNoSpaceDim();
-
+  size_t nf     = scalarOnly ? 1 : this->getNoFields();
+  size_t nVcomp = nf > this->getNoSpaceDim() ? this->getNoSpaceDim() : nf;
   bool haveXsol = false;
   if (mySol)
   {
-    if (scalarEq)
+    if (nf == 1)
       haveXsol = mySol->getScalarSol() != nullptr;
     else
       haveXsol = mySol->getVectorSol() != nullptr;
   }
+  if (myProblem && nf > 1)
+    nf = myProblem->getNoFields(1);
 
-  size_t nf = scalarEq ? 1 : this->getNoFields();
-  size_t pMAX = haveXsol ? nf+nf : nf;
-  std::vector<IntVec> sID(pMAX);
   std::array<IntVec,2> vID;
+  std::vector<IntVec> sID;
+  sID.reserve(haveXsol ? nf+nf : nf);
+
   Matrix field;
   Vector lovec;
 
-  size_t i, j, k;
   int geomID = myGeomID;
-  for (i = 0; i < myModel.size(); i++)
+  for (ASMbase* pch : myModel)
```

Comment:
can this be const ?

Reply:
I guess it can, since only const member functions are invoked for the pointed-to object.
Changed.